dart/lib/src/transport/transport.dart

```diff
@@ -0,0 +1,122 @@
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:meta/meta.dart';
+
+import '../protocol.dart';
+import '../sentry_options.dart';
+import '../utils.dart';
+import 'noop_encode.dart' if (dart.library.io) 'encode.dart';
+
+/// A transport is in charge of sending the event to the Sentry server.
+class Transport {
+  final SentryOptions _options;
+
+  @visibleForTesting
+  final Dsn dsn;
+
+  /// Use for browser stacktrace
+  final String _origin;
+
+  CredentialBuilder _credentialBuilder;
+
+  final Map<String, String> _headers;
+
+  Transport({@required SentryOptions options, String origin})
+      : _options = options,
+        _origin = origin,
+        dsn = Dsn.parse(options.dsn),
+        _headers = _buildHeaders(sdkIdentifier: options.sdk.identifier) {
+    _credentialBuilder = CredentialBuilder(
+      dsn: Dsn.parse(options.dsn),
+      clientId: options.sdk.identifier,
+      clock: options.clock,
+    );
+  }
+
+  Future<SentryId> send(SentryEvent event) async {
+    final data = event.toJson(origin: _origin);
+
+    final body = _bodyEncoder(
+      data,
+      _headers,
+      compressPayload: _options.compressPayload,
+    );
+
+    final response = await _options.httpClient.post(
+      dsn.postUri,
+      headers: _credentialBuilder.configure(_headers),
+      body: body,
+    );
+
+    if (response.statusCode != 200) {
+      return SentryId.empty();
+    }
+
+    final eventId = json.decode(response.body)['id'];
```

Comment:
Does this work on the envelope endpoint?

Reply:
we don't use the envelope endpoint, but /store, the envelope is part of the next version

Reply:
https://github.com/getsentry/sentry-dart/projects/1#card-48213613